arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2692,6 +2692,22 @@ void RestReplicationHandler::handleCommandHoldReadLockCollection() {
     return;
   }
 
+  // check if we are not the actual leader anymore. this can
+  // happen if there is a leader change after a follower scheduled a
+  // synchronize shard job
+  bool isLeader = col->followers()->getLeader().empty();
+  if (!isLeader) {
+    auto res = cancelBlockingTransaction(id);
```

Comment:
Would this transaction block the current server (now follower) from starting a SynchronizeShard with the new leader?